opm/parser/eclipse/EclipseState/Tables/TableContainer.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+  Copyright 2015 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_TABLE_CONTAINER_HPP
+#define OPM_TABLE_CONTAINER_HPP
+
+#include <cstddef>
+#include <memory>
+
+#include <opm/parser/eclipse/EclipseState/Tables/SimpleTable.hpp>
+
+
+namespace Opm {
+
+    class TableContainer {
+        /*
+          The TableContainer class implements a simple map:
+          {tableNumber , Table}. The main functionality of the
+          TableContainer class is that the getTable method implements
+          the Eclipse behavior:
+
+             If table N is not implemented - use table N - 1.
+
+          The getTable() method will eventually throw an exception if
+          not even table 0 is there.
+        */
+    public:
+        TableContainer( size_t maxTables );
+        bool empty() const;
+        size_t size() const;
+        void addTable(size_t tableNumber , std::shared_ptr<const SimpleTable> table);
+        size_t hasTable(size_t tableNumber) const;
```

Comment:
I would appreciate a comment about how hasTable() works: will it return false only if a call to getTable() with the same argument would fail? Or is it ok to call getTable(7) even if hasTable(7) is false (getting table 6 instead etc.).

Also, are typical table sets consecutive and indexed from 0 or 1? Are there table sets that typically have nonconsecutive numbers?